src/test/java/io/jenkins/plugins/coverage/model/CoverageSourceCodeRenderingAndCopyingITest.java

```diff
@@ -0,0 +1,136 @@
+import hudson.model.Run;
+import hudson.plugins.sshslaves.SSHLauncher;
+import hudson.slaves.DumbSlave;
+import hudson.slaves.EnvironmentVariablesNodeProperty;
+import hudson.slaves.EnvironmentVariablesNodeProperty.Entry;
+
+import io.jenkins.plugins.util.IntegrationTestWithJenkinsPerSuite;
+
+import static org.assertj.core.api.Assertions.*;
+import static org.assertj.core.api.Assumptions.*;
+
+/**
+ * Tests if source code Rendering and copying works.
+ */
+public class CoverageSourceCodeRenderingAndCopyingITest extends IntegrationTestWithJenkinsPerSuite {
+    private static final String JACOCO_FILE_NAME = "jacoco-analysis-model.xml";
+    private static final String COMMIT = "6bd346bbcc9779467ce657b2618ab11e38e28c2c";
+    private static final String REPOSITORY = "https://github.com/jenkinsci/analysis-model.git";
+    /** Docker container for java-maven builds. Contains also git to check out from an SCM. */
+    @Rule
+    public DockerRule<JavaGitContainer> javaDockerRule = new DockerRule<>(JavaGitContainer.class);
+
+    /** Example integration test for a pipeline with code coverage that runs on an agent. */
+    @Test
+    public void coveragePipelineOnAgentNode() throws IOException, InterruptedException {
+        assumeThat(isWindows()).as("Running on Windows").isFalse();
+
+        DumbSlave agent = createDockerContainerAgent(javaDockerRule.get());
+        WorkflowJob project = createPipelineOnAgent();
+
+        copySingleFileToAgentWorkspace(agent, project, JACOCO_FILE_NAME, JACOCO_FILE_NAME);
+
+        verifyGitRepository(project);
+
+    }
+
+    /**
+     * Reads source code from git and adds it to project.
+     */
+    @Test
+    public void sourceCodeCopyingTest() {
+        WorkflowJob job = createPipelineWithWorkspaceFiles(JACOCO_FILE_NAME);
+        job.setDefinition(new CpsFlowDefinition("node {"
+                + "    checkout([$class: 'GitSCM', "
+                + "branches: [[name: '" + COMMIT + "' ]],\n"
+                + "userRemoteConfigs: [[url: '" + REPOSITORY + "']],\n"
+                + "extensions: [[$class: 'RelativeTargetDirectory', \n"
+                + "            relativeTargetDir: 'checkout']]])\n"
+                + "    publishCoverage adapters: [jacocoAdapter('" + JACOCO_FILE_NAME
+                + "')], sourceFileResolver: sourceFiles('STORE_ALL_BUILD')\n"
+                + "}", true));
+
+        verifyGitRepository(job);
+    }
+
+    /**
+     * Verifies clone of repository by checking console log.
+     *
+     * @param workflowJob
+     *         job of the project
+     */
+    private void verifyGitRepository(final WorkflowJob workflowJob) {
+        Run<?, ?> build = buildSuccessfully(workflowJob);
+
+        String consoleLog = getConsoleLog(build);
+
+        assertThat(consoleLog)
+                .contains("Cloning repository " + REPOSITORY)
+                .contains("Checking out Revision " + COMMIT)
+                .contains("git checkout -f " + COMMIT);
+    }
```

Comment:
This verifies the Git cloning but what we need is verifying the source code copying and visualization